frontend/src/components/App/settings/index.tsx

```diff
@@ -0,0 +1,188 @@
+  const { defaultValue } = props;
+  const [isSelectOpen, setIsSelectOpen] = useState(false);
+  const [options, setOptions] = useState(defaultValue);
+  const [selectedValue, setSelectedValue] = useState(DefaultRowsPerPageOptions[0]);
+  const [customValue, setCustomValue] = useState<number | undefined>(undefined);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(setSettingsTableRowsPerPageOptions(options));
+  }, [options]);
+
+  const handleChange = (
+    event: ChangeEvent<{
+      name?: string | undefined;
+      value: unknown;
+    }>
+  ) => {
+    const val = parseInt(event.target.value as string);
+    setSelectedValue(val);
+  };
+
+  const handleClose = () => {
+    setIsSelectOpen(false);
+  };
+
+  const handleOpen = () => {
+    setIsSelectOpen(true);
+  };
+
+  return (
+    // we have assigned value -1 to select custom option
+    selectedValue === -1 ? (
+      <Box display={'flex'}>
+        <Box mr={1} width="60vw">
+          <TextField
+            fullWidth
+            type={'number'}
+            value={customValue}
+            error
+            placeholder={t('Enter a custom row value')}
+            onChange={e => {
+              const val = parseInt(e.target.value);
+              if (val > DefaultRowsPerPageOptions[DefaultRowsPerPageOptions.length - 1]) {
+                setCustomValue(parseInt(e.target.value));
+              }
+            }}
+          />
+        </Box>
+        <Button
+          variant="contained"
+          onClick={() => {
+            if (customValue === undefined) {
+              return;
+            }
+            // remove end element and add custom value
+            let newOptions;
+            if (
+              options[options.length - 1] ===
+              DefaultRowsPerPageOptions[DefaultRowsPerPageOptions.length - 1]
+            ) {
+              newOptions = [...options];
+            } else {
+              newOptions = options.slice(0, options.length - 1);
+            }
+            newOptions.push(customValue as number);
+            setOptions(newOptions);
+            setSelectedValue(customValue as number);
+          }}
+        >
+          Add
+        </Button>
+      </Box>
+    ) : (
+      <FormControl fullWidth>
+        <Select
+          value={selectedValue}
+          style={{}}
+          open={isSelectOpen}
+          onClose={handleClose}
+          onOpen={handleOpen}
+          onChange={handleChange}
+        >
+          {options.map(option => {
+            return (
+              <MenuItem key={option} value={option}>
+                {option}
+              </MenuItem>
+            );
+          })}
+          <MenuItem key={'custom'} value={-1}>
+            {t('custom')}
+          </MenuItem>
+        </Select>
+      </FormControl>
+    )
+  );
+}
+
+export default function Settings() {
+  const { t } = useTranslation(['settings']);
+  const settingsObj = JSON.parse(localStorage.getItem('settings') || '{}');
```

Comment:
The settings here should come from the redux state, not the localstorage.

Reply:
fixed